app/bundles/LeadBundle/Segment/ContactSegmentService.php

```diff
@@ -361,7 +361,7 @@ private function timedFetchAll(QueryBuilder $qb, $segmentId)
     {
         try {
             $start  = microtime(true);
-            $result = $qb->execute()->fetchAll(\PDO::FETCH_ASSOC);
+            $result = $qb->execute()->fetchAll();
```

Comment:
This is not replacing the deprecated method, right?

Reply:
Yes, but when I change it to `fetchAllAssociative`, `PHPStan` shows [an error ](https://github.com/mautic/mautic/actions/runs/4723025209/jobs/8378416839?pr=12218)

Reply:
I see. This one is tricky. I looked around and it seem to me that the cleanest way how to fix this is to add this new method to the Segment QueryBuilder:
```php
public function fetchAllAssociative(): array
    {
        return $this->connection->fetchAllAssociative($this->getSQL(), $this->params, $this->paramTypes);
    }
```

And then use it here:
```php
$result = $qb->fetchAllAssociative();
```
WDYT?

Reply:
@escopecz taking a closer look into the code, the code we have 

```
    /**
     * Executes this query using the bound parameters and their types.
     *
     * Uses {@see Connection::executeQuery} for select statements and {@see Connection::executeUpdate}
     * for insert, update and delete statements.
     *
     * @return Result|int|string
     *
     * @throws \Doctrine\DBAL\DBALException
     */
    public function execute()
    {
        if (self::SELECT == $this->type) {
            return $this->connection->executeQuery($this->getSQL(), $this->params, $this->paramTypes);
        } else {
            return $this->connection->executeStatement($this->getSQL(), $this->params, $this->paramTypes);
        }
    }
```

Is just an old copy of `Doctrine\DBAL\Query\QueryBuilder` has

```
    /**
     * Executes this query using the bound parameters and their types.
     *
     * @return ForwardCompatibility\Result|int|string
     *
     * @throws Exception
     */
    public function execute()
    {
        if ($this->type === self::SELECT) {
            return ForwardCompatibility\Result::ensure(
                $this->connection->executeQuery($this->getSQL(), $this->params, $this->paramTypes)
            );
        }

        return $this->connection->executeStatement($this->getSQL(), $this->params, $this->paramTypes);
    }
```

I just removed our `execute` method and everything is working as expected

Reply:
Solved in commit 06b11f80bec21880218bdbc7b436020d67e22c6c

Reply:
Wow, that's interesting. Great solution!